files/borgback-adfc-final.sh

```diff
@@ -0,0 +1,116 @@
+REPO_NAME="name-des-borg-repositories"               # Name des Backup-Repositories auf Backup-Client
+ARCHIVE_NAME="name-des-archivs_${TIMESTAMP}"         # Name der aktuellen Sicherung
+BORG_USER="borguser"                                 # Benutzerkonto auf Backup-Client / FIX! NICHT ÄNDERN
+
+PRUNE_D="7"                                          # Anzahl der vorzuhaltenden Tagessicherungen
+PRUNE_W="4"                                          # Anzahl der vorzuhaltenden Wochensicherungen
+PRUNE_M="12"                                         # Anzahl der vorzuhaltenden Monatssicherungen
+PRUNE_Y="2"                                          # Anzahl der vorzuhaltenden Jahressicherungen
+
+M_SND="absender@e-mail.tld"                          # Absender
+M_RCV="empfaenger@e-mail.tld"                        # Empfänger
+M_PWD="smtp-passwort"                                # SMTP-Passwort
+M_SVR="mailserver.hoster.tld"                        # SMTP-Server
+M_SUB="Backup-Report"                                # Betreffzeile / sollte so bleiben
+M_BDY="Freier Text für E-Mail-Body."                 # Body Text
+M_ATT="anhang1.ext /pfad/zum/anhang2.ext"            # weitere Anhänge hier eintragen (Backup-Logdatei geht standardmäßig immer mit raus)
+
+
+
+###########################################################################################
+############ AB HIER KEINE ANPASSUNGEN MEHR ERFORDERLICH! #################################
+###########################################################################################
+
+
+# ---------- Einlesen der verbundenen Backup-Clients / Ermitteln der IP -------------------
+
+# Ist kein Client verbunden wird auch nichts unternommen
+if [ "$(ls -A $CL_ONLINE_DIR)" ]; then
+   MACHE="WEITER"
+else
+   exit 1
+fi
+
+# Einlesen aller Verbindungen
+ALLCLIENTS=()
+for FILES in $CL_ONLINE_DIR/*; do
+    ALLCLIENTS[${#ALLCLIENTS[@]}]="$FILES"
+done
+
+# Alle in CL_ONLINE_DIR vorhandenen Verbindungen testen, wenn Offline löschen. Nur 192er-Netz zulässig
+for FILES in "${ALLCLIENTS[@]}"; do
+    TEST_IP=$(echo "$FILES" | cut -d '=' -f 2)
+    if [ $(echo "$TEST_IP" | cut -d '.' -f 1) -ne 192 ]; then
+       rm $FILES
+    fi
+    ping -c 3 $TEST_IP 2>&1> /dev/null
+    RESULT=$?
+    if [ $RESULT == 0 ]; then
+       IPADDR=$TEST_IP
+       ONLINE_FILE=$FILES
+    else
+       rm $FILES
+    fi
+done
+
+# Falls keine Verbindung funktionierte ist das Verzeinis leer. Dann Abbruch!
+if [ "$(ls -A $CL_ONLINE_DIR)" ]; then
+   MACHE="WEITER"
+else
+   exit 2
+fi
+
+
+
+# ---------- Backup-Routine ----------------------------------------------------------------
+
+# Repo-Init legt beim erstmaligen Aufruf ein Repository an. Weitere Aufrufe beenden sich ohne Funktion oder Fehler (Rückgabewert=2 spielt hier keine Rolle).
+BORG_PASSPHRASE="$SECRET" borg init --encryption=repokey --show-rc $BORG_USER@$IPADDR:$REPO_PATH/$REPO_NAME 2>&1
```

Comment:
Das Passwort können wir uns IMHO sparen. Es ist sowieso auf dem Master bekannt.

Reply:
Siehe die Ausführungen hierzu in meiner Mail: Mit dem Passwort schaffen wir eine zusätzliche Schutzmauer bei Diebstahl eines Backup-Clients (sofern ich die Doku zu Borg richtig interpretiere). Schlüssel und Passwort liegen getrennt und sich für sich allein wertlos.

Reply:
Aber das Passwort ist genau so geschützt wie der Key. Wenn der Angreifer zugriff auf den Key hat, hat er auch das Passwort und umgekehrt. Das ist nur gefühlte Sicherheit.